lib/db.js

```diff
@@ -92,24 +92,45 @@ class LimitDBRedis extends EventEmitter {
     this.buckets[key] = utils.buildBucket(bucket);
   }
 
-  bucketKeyConfig(type, key) {
-    const fromCache = type.overridesCache.get(key);
+  paramsToBucketConfig(params) {
+    const bucketConfig = {};
+
+    if (params && params.configOverride) {
+      if (params.configOverride.bucketSize) {
+        bucketConfig.size = params.configOverride.bucketSize;
+      }
+      if (params.configOverride.refillRateMsPerToken) {
+        bucketConfig.per_interval = 1;
+        bucketConfig.interval = params.configOverride.refillRateMsPerToken;
+      }
+    }
+
+    return bucketConfig;
+  }
+
+  /**
+   * @param {string} type
+   * @param {object} params
+   * @returns
+   */
+  bucketKeyConfig(type, params) {
+    const fromCache = type.overridesCache.get(params.key);
```

Comment:
My expectation was we wouldn't need to merge a result, and instead a specific size/refillRate would be passed instead.

In other words, if those parameters are provided, bucket configuration is fully ignored (is a configuration-less mode).

But seems the fact buckets are pre-defined is quite ingrained in the codebase, so I'm OK with the approach 👍 .

Reply:
Let me know if you'd rather me switch, I can easily switch it over to either/or 🤷‍♂️

Reply:
I agree that merging is probably unexpected and that if params are passed, static bucket configs should be ignored completely. Seems less surprising, and even slightly more efficient.

Reply:
Sounds like the consensus is either/or then.  I'll make that switch 👍

Reply:
Merging blindly is also risky: since some attributes are optional, you could end up with some attributes from the `configOverride` and some attributes from the static config.

One example of surprising behavior:
- static config has a `per_interval` value (auto-refill)
- `configOverride` only passes a value of `{size:10}` with the intention of disabling auto-refill

The resulting object contains `{per_interval: xxx, size: 10}`, so we unexpectedly end up with an auto-refilling config.

Reply:
Indeed, totally get it 👍 .